scripts/nbd_client_manager.py

```diff
@@ -123,7 +127,14 @@ def _find_unused_nbd_device():
 
 
 def _wait_for_nbd_device(nbd_device, connected):
+    deadline = datetime.now() + timedelta(minutes=MAX_DEVICE_WAIT_MINUTES)
```

Comment:
would be good to use a monotonic clock here, otherwise when the host's clock is changed this could still hang for a long time: https://docs.python.org/3/library/time.html#time.monotonic.
Although that might have to wait until this script is converted to Python3 (if it isn't already), or until we no longer need this script (e.g. because we make vhd-util talk NBD directly).

Reply:
We do have monotonic (as SM uses it) but the likelihood of it hitting the issue is so low (in the absence) of the Kernel bug that it's not worthing stressing over.